crypto/ml_dsa/ml_dsa_key.c

```diff
@@ -421,50 +439,51 @@ static int keygen_internal(const uint8_t *seed, size_t seed_len,
         && shake_xof(md_ctx, out->shake256_md, out->pub_encoding, out->params->pk_len,
                      out->tr, sizeof(out->tr))
         && ossl_ml_dsa_sk_encode(out);
+
 err:
+    if (out->seed != NULL && !out->retain_seed) {
+        OPENSSL_cleanse(out->seed, ML_DSA_SEED_BYTES);
```

Comment:
OPENSSL_clear_free()

Reply:
Sure, coming up. In GSSAPI, you free pointers by reference, and they're also set to NULL for you, here I still have to do that explicitly.